File: src/data/courses/dsa/queues/deque/sliding-window-maximum/code/java/SlidingWindowMaximumSol1.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int[] res = new int[nums.length-k+1];
        for (int i=0;i<=nums.length-k;i++) {
            int max = nums[i]; for(int j=i+1;j<i+k;j++) max=Math.max(max,nums[j]);
            res[i]=max;
        } return res;
    }
}

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/sliding-window-maximum/code/cpp/sliding_window_maximum_sol1.cpp
class Solution {
public:
    vector<int> maxSlidingWindow(vector<int>& nums, int k) {
        vector<int> result;
        for (int i = 0; i <= (int)nums.size()-k; i++)
            result.push_back(*max_element(nums.begin()+i, nums.begin()+i+k));
        return result;
    }
};